src/cupbearer/data/backdoors.py

```diff
@@ -106,6 +107,27 @@ def __post_init__(self):
         p_transform = self.p_backdoor + self.p_noise
         assert 0 <= p_transform <= 1, "Probability must be between 0 and 1"
 
+    @staticmethod
+    def _get_savefile_fullpath(basepath):
+        return os.path.join(basepath, "wanet_backdoor.npy")
+
+    def store(self, basepath):
+        super().store(basepath)
+        # TODO If img size is known the transform can be significantly sped up
```

Comment:
It would require knowing the size of the image before starting training, but I figure I could just make it a lazy property and then store and load the full field.

Reply:
Only calculating the full warping field once is the key thing for speed-up. So I guess storing and loading the 4x4 warping matrices is fine if we want to keep all doors open.

Reply:
Yeah, I don't think it matters much which one we store on disk (I guess storing the full field has the advantage that it also deals with the `control_grid_width` thread below, but that's easy to fix anyway).

Agree that computing the field once we get the first image and then caching is the best way to do this